crates/h264/src/sps.rs

```diff
@@ -549,6 +550,198 @@ impl Sps {
         })
     }
 
+    /// Builds the Sps struct into a byte stream.
+    /// Returns a built byte stream.
+    pub fn build<T: io::Write>(&self, writer: &mut BitWriter<T>) -> io::Result<()> {
```

Comment:
This should not take in a bitwriter. It should just be a `std::io::Write` we should then construct it using a bitwriter and then add the emulation prevention byte when 3 zero bytes are together.